sktime/transformations/series/summarize.py

```diff
@@ -750,3 +750,140 @@ def get_test_params(cls, parameter_set="default"):
+        "y_inner_mtype": "None",  # which mtypes do _fit/_predict support for X?
+        "fit_is_empty": True,
+    }
+
+    def __init__(self, transformer, splitter, index="last"):
+        self.transformer = transformer
+        self.splitter = splitter
+        self.index = index
+
+        super().__init__()
+
+    def _transform(self, X, y=None):
+        """Transform X and return a transformed version.
+
+        Transforms the series `X` into a new series `Xt` by applying the
+        series-to-primitives transformer `t` on each train split of `X`,
+         which is created using the splitter `cv`.
+
+         The `i-th` row of `Xt` is equivalent to
+         `t.fit_transform(cv.split_series(X)[i][0])`.
+
+        private _transform containing the core logic, called from transform
+
+        Parameters
+        ----------
+        X : pd.Series or pd.DataFrame
+            Data to be transformed
+        y : ignored argument for interface compatibility
+            Additional data, e.g., labels for transformation
+
+        Returns
+        -------
+        Xt : pd.DataFrame
+            The transformed Data
+        """
+        if not hasattr(self.transformer, "fit_transform"):
+            raise ValueError("Transformer should have a fit_transform method")
+        if not hasattr(self.splitter, "split_series"):
+            raise ValueError("Splitter should have a split_series method")
+
+        transformed_series = []
+        splits = self.splitter.split_series(X)
+
+        for split in splits:
+            tf = self.transformer.clone()
+            transformed_split = tf.fit_transform(split[0])
+            transformed_split.index = [split[0].index[-1]]
+            transformed_series.append(transformed_split)
+
+        Xt = pd.concat(transformed_series)
+
+        if self.index != "last":
+            Xt = Xt.reset_index(drop=True)
+
+        return Xt
+
+    @classmethod
+    def get_test_params(cls, parameter_set="default"):
+        """Return testing parameter settings for the transformer.
+
+        Parameters
+        ----------
+        parameter_set : str, default="default"
+            Name of the set of test parameters to return, for use in tests. If no
+            special parameters are defined for a value, will return `"default"` set.
+
+        Returns
+        -------
+        params : dict or list of dict, default = {}
+            Parameters to create testing instances of the class
+            Each dict are parameters to construct an "interesting" test instance, i.e.,
+            `MyClass(**params)` or `MyClass(**params[i])` creates a valid test instance.
+            `create_test_instance` uses the first (or only) dictionary in `params`
+        """
+        from sktime.forecasting.model_selection import ExpandingWindowSplitter
+
+        params1 = {
+            "transformer": SummaryTransformer(),
+            "splitter": ExpandingWindowSplitter(),
+        }
+
```

Comment:
Generally, it's a good idea to have at least two test examples.

You should try a different setting for `index`, and maybe another combination of transformer and splitter.

A good idea is to cover transformers that need to be fitted..

Reply:
yes, of course - this is feature creep, so a potential separate project.